microsoft-edge/extensions-chromium/developer-guide/sidebar.md

```diff
@@ -0,0 +1,285 @@
+
+`manifest.json`:
+
+```json
+{
+  "name": "My sidebar extension",
+  ...
+   "action": {
+    "default_title": "Click to open panel"
+  },
+  ...
+}
+```
+
+`service-worker.js`:
+
+```js
+const BING_ORIGIN = 'https://www.bing.com';
+
+// Allows users to open the side panel by clicking on the action toolbar icon
+chrome.sidePanel
+  .setPanelBehavior({ openPanelOnActionClick: true })
+  .catch((error) => console.error(error));
+...
+```
+<!-- ------------------------------ -->
+
+#### Example 4: Switching to a different sidebar
+
+An extension can enable different sidebar for a specific tab.  
+
+The following example sets a sidebar with welcome message on runtime.OnInstalled(). When the user navigates to a different tab, the sidebar is replaced with the browser level sidebar.
+
+`service-worker.js`:
+
+```js
+const welcomePage = 'sidebar/welcome-sp.html';
+const mainPage = 'sidebar/main-sp.html';
+
+chrome.runtime.onInstalled.addListener(() => {
+  chrome.sidePanel.setOptions({ path: welcomePage });
+});
+
+chrome.tabs.onActivated.addListener(async ({ tabId }) => {
+  const { path } = await chrome.sidePanel.getOptions({ tabId });
+  if (path === welcomePage) {
+    chrome.sidePanel.setOptions({ path: mainPage });
+  }
+});
+```
+<!-- ------------------------------ -->
+
+<!-- ====================================================================== -->
+## User experience for extensions in Edge sidebar 
+
+
+<!-- ------------------------------ -->
+#### Identifying   Extensions with Sidebar support
+
+* The sidebar icon is displayed next to the extension name in the hub in the toolbar:
+
+![Sidebar dialog](./sidebar-images/sidebar-dialog.png)
+<!-- todo: put real content in png -->
+
+* On the Microsoft Edge Add-Ons store page, there's a dialog on the extension product page, highlighting the sidebar feature.
+<!-- todo: show that "dialog", is it actually a dialog? -->
+
+
+<!-- ------------------------------ -->
+#### Opening Extensions in the Sidebar
+
+To open an extension in the sidebar, the user can do either of the following:
+
+* Click the sidebar icon in the hub in the toolbar.
+
+* Right-click the extension's shortcut in the toolbar.
+
+
+<!-- ====================================================================== -->
+## Properties for extensions in Sidebar   
+
+#### GetPanelOptions
+
+| Property | Description |
+|---|---|
+| `tabId`  
+`number optional` | If specified, the side panel options for the given tab will be returned. Otherwise, returns the default side panel options (used for any tab that doesn't have specific settings).|
+
+<!-- ------------------------------ -->
+#### Panel behavior
+
+| Property | Description |
+|---|---|
+| `openPanelOnActionClick` 
+`boolean optional` | Whether clicking the extension's icon toggles showing the extension's entry in the sidebar.  The default value is `false`. |
+
+
+<!-- ------------------------------ -->
+#### Panel options
+
```

Comment:
there's a build error here, i guess you might change to 3-col table, such as:
```
| Property | Description | Optional? |
|---|---|---|
| text | text | text |
```

Reply:
@Bhuvana-Priya - one of the few markup supported in table cell is <br/>, html for manual line break

Reply:
completely reformatted